kin-recovery/README.md

```diff
@@ -0,0 +1,180 @@
+...
+        maven {
+            url 'https://jitpack.io'
+        }
+    }
+}
+```
+###### Snippet: Modifying module build files
+
+```gradle
+...
+dependencies {
+    ...
+    implementation 'com.github.kinecosystem.kin-sdk-android:recovery:<latest release>'
+}
+```
+
+For the latest release version, go to [https://github.com/kinecosystem/kin-sdk-android/releases](https://github.com/kinecosystem/kin-sdk-android/releases).
+
+See the main repository at [github.com/kinecosystem/kin-sdk-android](https://github.com/kinecosystem/kin-sdk-android).
+
+
+## Overview
+
+Launching Backup and Restore flows requires the following steps:
+
+1. Creating the backup and restore manager
+2. Adding callbacks
+3. Passing the result
+4. Backup or Restore
+
+### Step 1 - Creating the Backup and Restore Manager
+
+First you will need to have a KinClient object for later use when you will invoke backup and/or restore methods.  
+Also you need to create the BackupAndRestoreManager object.
+###### Snippet: This is an example of how to create those objects:
+
+```java
+kinClient = new KinClient(context, Environment.TEST, "1acd")
+...
+backupAndRestoreManager = new BackupAndRestoreManager(context);
+```
+
+
+For more details on KinClient, see https://github.com/kinecosystem/kin-sdk-android/tree/master/kin-sdk.
+
+### Step 2 - Adding Callbacks
+
+Both callbacks have 3 methods:
+ - `onSuccess` is called when the operation in finished successfully and in restore it also have a `KinAccount` object which is the restored account.
+- `onCancel` is called when the user is going back from the backup and restore activty to your activity.
+- `onError()` is called if there was some error in the process of backup or restore.
+
+```java
+// For backup
+backupAndRestoreManager.registerBackupCallback(new BackupCallback() {
+    @Override
+    public void onSuccess() {
+        // here you can handle the success.
+    }
+    
+    @Override
+    public void onCancel() {
+        // here you can handle the cancellation.
+    }
+    
+    @Override
+    public void onFailure(BackupException throwable) {
+        // here you can handle the failure.
+    }
+});
+
+// For restore  
+backupAndRestoreManager.registerRestoreCallback(new RestoreCallback() {
+    @Override
+    public void onSuccess(KinAccount kinAccount) {
+        // here you can handle the success.
+    }
+    
+    @Override
+    public void onCancel() {
+        // here you can handle the cancellation.
+    }
+    
+    @Override
+    public void onFailure(BackupException throwable) {
+        // here you can handle the failure.
+    }
+});
+```
+##### NOTE:
+Be sure to unregister from the module when no longer needed.  
+When you want to unregister then please use release:
+###### Snippet: unregister from the module and release all its resources
+```java 
+backupAndRestoreManager.release();
+``` 
+We recommend to register/unregister the callbacks in a way that will “survive” activity restart or similar situations.  
```

Comment:
I think it should be definitive, registering callbacks should happen in `onCreate` (you can add in order to catch `onActivityResult` internally by the SDK)
the mention of MVP is irrelevant I think, because what's matter is Android lifecycle.

Reply:
I already remove the MVP stuff.
Will remove also the "recommended"